### DISCUSSION TITLE
What's up with DRONE_* envs?

### DISCUSSION THREAD

**Discussion by zc-devs** [woodpecker-ci/woodpecker#4098] [Ask a Question] [answered]
#3846:

> I would add DRONE_* to the list.

> We keep the drone vars to be compatible to drone plugins.

https://woodpecker-ci.org/docs/migrations

0.15

> Dropped support for DRONE_* environment variables. The according WOODPECKER_* variables must be used instead.

> Dropped support for DRONE_* environment variables in pipeline steps. Pipeline meta-data can be accessed with CI_* variables.

So, where is the truth?

Is there some docs, stating, that it was brought back for compatibility?

**Comment by 6543** [comment, marked as answer]
We dropped the DRONE_ vars as **official** build-in vars.
But we never removed them completely so drone plugins worked as the did.

nowadays we have a dedicated "compatibility" function that do provide this feature and also works with quirks we know of:

https://github.com/woodpecker-ci/woodpecker/blob/main/pipeline/frontend/metadata/drone_compatibility.go

As for the docs we intentionally don't mention them, as users should not use them. But we do document that drone plugins can be used just natively.


I hope that covers all questions?
